docs/idm-concept.md

```diff
@@ -0,0 +1,206 @@
+
+The way a public key is declared as revoked is [yet to be standardized](https://github.com/w3c-ccg/did-spec/issues/96) in the DID spec. Nevertheless, a DID-Document may be augmented via the `@context` attribute to support having that information.
+
+#### Replication and synchronization
+
+The same identity might be held inside different IDM Wallets. For that reason, the identities' data, such as its credentials and applications, will be seamlessly replicated to other IDM Wallets using P2P technologies powered by [CRDTs](https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type).
+
+Any data stored off-chain or on-chain as per the DID-method best-practices will be taken into consideration, meaning they will be synchronized accordingly.
+
+### IDM Wallet UI
+
+End users need a graphical user-interface to interact with their IDM Wallet. An IDM Wallet UI is an application that will make use of the most suitable IDM Wallet SDK for the programming language they choose.
+
+Even though anyone can implement its own IDM Wallet UI application, there is a need to define some standards in regards to user journeys, user experience and interactions. For that reason, there will be a reference IDM Wallet UI based on web-technologies to be used inside a browser, that can be packed as an [Electron](https://electronjs.org/) in the future.
+
+#### Reference UI diagrams
+
+User-journeys diagram:
+
+<img src="https://raw.githubusercontent.com/ipfs-shipyard/pm-idm/master/docs/images/diagram_user-journeys.png" alt="" width="1000">
+
+Information Architecture diagram:
+
+<img src="https://raw.githubusercontent.com/ipfs-shipyard/pm-idm/master/docs/images/diagram_information-architecture.png" alt="" width="1000">
+
+#### Reference UI concept
+
+The goal of the reference UI is to have a really polished interface with a premium feel. It must be easy to use, distinct, reliable, accessible and provide a feel of credibility to the users.
+
+<img src="https://raw.githubusercontent.com/ipfs-shipyard/pm-idm/master/docs/images/ui-concept_setup-locker.png" alt="" width="1000">
+<img src="https://raw.githubusercontent.com/ipfs-shipyard/pm-idm/master/docs/images/ui-concept_homepage.png" alt="" width="1000">
+<img src="https://raw.githubusercontent.com/ipfs-shipyard/pm-idm/master/docs/images/ui-concept_profile-page.png" alt="" width="1000">
+
+### IDM Client
+
+Applications need to interact with the IDM Wallet. This process will be facilitated by the IDM Client, which will be available as multiple SDKs for different programming languages. Each SDK will provide a simple and intuitive interface based on the [IDM Client Spec](./spec/idm-spec#idm-client) and contain means to authenticate, unauthenticate, sign, verify signatures, amongst others.
+
+There will be a reference IDM Client written in JavaScript, suitable to use inside a browser.

```

Comment:
It will be nice to expand on this idea and fully discuss the ramifications of crypto in `content` documents. I was always imagining this crypto being exposed inside IPFS-published sites (or stand-alone IPFS electron apps) where the content is immutable & where browsers can consume (or use P2P methods / APIs) crypto code with a bit more built-in safety.

Reply:
Are you referring to how developers are expected to sign documents and verify their signatures?

- The verification of signatures are detailed in the `Verifying signatures` section.
- The IDM Client will offer two functions, one for signing and one for verifying signatures. These will be detailed in the IDM spec paper which I will work next.

Perhaps we could refer to them in this section?

On the other hand, if I misunderstood your comment, could you elaborate a bit on what you said?

Reply:
Perhaps I was not very clear. The very concept of performing cryptographic activities in a web page - with code delivered from a server - is a dangerous thing. How can we be sure the code delivered to the client has not been changed by an attacker? SSL only gets one so far. A packaged application will always be preferred to security auditors, especially apps that can be verified. I believe this is a good paper on the subject: https://madiba.encs.concordia.ca/~x_decarn/papers/verifiable-build-acsac2014.pdf

Naturally, IPFS-delivered Dapps give us many properties of `"verifiable builds"`, but web server hosted apps do not.

Reply:
Oh, I understood now. 

The IDM Client will be available through various SDKs that may be distributed in a various of ways: from a naive downloaded link to a registry, such `npm`.

Assuming that developers already evaluated a IDM Client codebase that they want to use ( including each time they update), there must be some sort of checksum involved. Usually, linux distributions make checksums public for developers to check against what they downloaded. AFAIK, `npm` already does this via the checksums on the `package-lock` file. I think it really depends on how and where the SDKs are published and securities those publish channels already give.

While I think this topic is somewhat out-of-scope, maybe it's worth briefly mentioning this in a short-paragraph. Want to take a shot on this @daviddahl?

Reply:
We may also use https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity specifically for the web.

Reply:
To not block this PR, I've created https://github.com/ipfs-shipyard/pm-idm/issues/39. @daviddahl feel free to take on that issue!